server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -974,6 +974,25 @@ public static <T extends Enum<T>> Parameter<T> enumParam(
             Function<FieldMapper, T> initializer,
             T defaultValue,
             Class<T> enumClass
+        ) {
+            return enumParam(name, updateable, initializer, (Supplier<T>) () -> defaultValue, enumClass);
```

Comment:
Added extra `Supplier<T>` related methods here, so we are aware of what version we're using (see https://github.com/elastic/elasticsearch/pull/98268/files#r1287069780)